go/yaml2/yaml_test.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2024 The Vitess Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package yaml2
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestYamlVars(t *testing.T) {
+	type TestStruct struct {
+		StringField  string  `yaml:"stringfield"`
+		IntField     int     `yaml:"intfield"`
+		BoolField    bool    `yaml:"boolfield"`
+		Float64Field float64 `yaml:"float64field"`
+	}
+
+	inputData := TestStruct{
+		"tricky text to test text",
+		32,
+		true,
+		3.141,
+	}
+
+	//testing Marshal
```

Comment:
Should also have a test per function here.

Reply:
Can you please elaborate? Would it be better to Marshal and Unmarshal for each struct field or assert and verify after marshaling and then doing same for unmarshaling?